pkg/k8s/utils/utils_test.go

```diff
@@ -113,8 +113,8 @@ func TestServiceEndpointsAndSlices(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to list services: %s", err)
 	}
-	if len(epSlices.Items) != 1 || epSlices.Items[0].ObjectMeta.Name != "test-svc-1" {
-		t.Fatalf("Expected test-svc-1, retrieved: %v", epSlices)
+	if len(epSlices.Items) != 2 || epSlices.Items[0].ObjectMeta.Name != "test-svc-1" || epSlices.Items[1].ObjectMeta.Name != "test-svc-2" {
```

Comment:
I'm not sure if the order of the EndpointSlices returned by List is guaranteed to be the same as the order in which they are created. If it's not, we risk to make this test flaky. I'd go for a set-like comparison, just in case.

Reply:
Actually I red again those tests and service proxy tests are handled in another test anyway so I ended up removing the case for the service proxy that I added and replacing that by the test about not reading the endpoint synced by the endpointslice mesh controller that was actually not tested.